bapm_server/bapm_consumer.py

```diff
@@ -0,0 +1,230 @@
+        # LOGGER.info('Queue bound: %s', userdata)
+        self.set_qos()
+
+    def set_qos(self):
+        self._channel.basic_qos(
+            prefetch_count=self._prefetch_count, callback=self.on_basic_qos_ok
+        )
+
+    def on_basic_qos_ok(self, _unused_frame):
+        # LOGGER.info('QOS set to: %d', self._prefetch_count)
+        self.start_consuming()
+
+    def start_consuming(self):
+        # LOGGER.info('Issuing consumer related RPC commands')
+        self.add_on_cancel_callback()
+        self._consumer_tag = self._channel.basic_consume(self.QUEUE, self.on_message)
+        self.was_consuming = True
+        self._consuming = True
+
+    def add_on_cancel_callback(self):
+        # LOGGER.info('Adding consumer cancellation callback')
+        self._channel.add_on_cancel_callback(self.on_consumer_cancelled)
+
+    def on_consumer_cancelled(self, method_frame):
+        # LOGGER.info('Consumer was cancelled remotely, shutting down: %r',
+        #             method_frame)
+        if self._channel:
+            self._channel.close()
+
+    def on_message(self, _unused_channel, basic_deliver, properties, body):
+        # LOGGER.info('Received message # %s from %s: %s',
+        #             basic_deliver.delivery_tag, properties.app_id, body)
+        print(
+            "Received message # %s from %s: %s",
+            basic_deliver.delivery_tag,
+            properties.app_id,
+            body,
+        )
+        self._thread_queue.put(body)
+        self.acknowledge_message(basic_deliver.delivery_tag)
+
+    def acknowledge_message(self, delivery_tag):
+        # LOGGER.info('Acknowledging message %s', delivery_tag)
+        self._channel.basic_ack(delivery_tag)
+
+    def stop_consuming(self):
+        if self._channel:
+            # LOGGER.info('Sending a Basic.Cancel RPC command to RabbitMQ')
+            cb = functools.partial(self.on_cancelok, userdata=self._consumer_tag)
+            self._channel.basic_cancel(self._consumer_tag, cb)
+
+    def on_cancelok(self, _unused_frame, userdata):
+        self._consuming = False
+        # LOGGER.info(
+        #     'RabbitMQ acknowledged the cancellation of the consumer: %s',
+        #     userdata)
+        self.close_channel()
+
+    def close_channel(self):
+        # LOGGER.info('Closing the channel')
+        self._channel.close()
+
+    def run(self):
+        self._connection = self.connect()
+        self._connection.ioloop.start()
+
+    def stop(self):
+        if not self._closing:
+            self._closing = True
+            LOGGER.info("Stopping")
+            if self._consuming:
+                self.stop_consuming()
+                self._connection.ioloop.start()
+            else:
+                self._connection.ioloop.stop()
+            LOGGER.info("Stopped")
+
+
+def main():
+    # logging.basicConfig(level=logging.DEBUG, format=LOG_FORMAT)
+    amqp_url = "amqp://guest:guest@aw-sdx-monitor.renci.org:5672/%2F"
```

Comment:
Can we get the `amqp_url` value from an env var or configuration?

Reply:
This `main` is usually for testing consumer only, the `bapm_consumer.py` should usually be used as a python module, so I feel it should be ok for now, I updated the other places to get values from environ variables.